src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -0,0 +1,94 @@
+import org.opensearch.ml.common.output.model.ModelTensors;
+
+/**
+ * This class will act as an abstraction on the MLCommons client for accessing the ML Capabilities
+ */
+@RequiredArgsConstructor
+@Log4j2
+public class MLCommonsClientAccessor {
+    private static final List<String> TARGET_RESPONSE_FILTERS = List.of("sentence_embedding");
+    private final MachineLearningNodeClient mlClient;
+
+    /**
+     * Abstraction to call predict function of api of MLClient with default targetResponse filters. It uses the
+     * custom model provided as modelId and run the {@link MLModelTaskType#TEXT_EMBEDDING}. The return will be sent
+     * using the actionListener which will have a {@link List} of {@link List} of {@link Float} in the order of
+     * inputText. We are not making this function generic enough to take any function or TaskType as currently we
+     * need to run only TextEmbedding tasks only.
+     *
+     * @param modelId {@link String}
+     * @param inputText {@link List} of {@link String} on which inference needs to happen
+     * @param listener {@link ActionListener} which will be called when prediction is completed or errored out
+     */
+    public void inferenceSentences(
+        @NonNull final String modelId,
+        @NonNull final List<String> inputText,
+        @NonNull final ActionListener<List<List<Float>>> listener
+    ) {
+        inferenceSentences(TARGET_RESPONSE_FILTERS, modelId, inputText, listener);
+    }
+
+    /**
+     * Abstraction to call predict function of api of MLClient with provided targetResponse filters. It uses the
+     * custom model provided as modelId and run the {@link MLModelTaskType#TEXT_EMBEDDING}. The return will be sent
+     * using the actionListener which will have a {@link List} of {@link List} of {@link Float} in the order of
+     * inputText. We are not making this function generic enough to take any function or TaskType as currently we
+     * need to run only TextEmbedding tasks only.
+     *
+     * @param targetResponseFilters {@link List} of {@link String} which filters out the responses
+     * @param modelId {@link String}
+     * @param inputText {@link List} of {@link String} on which inference needs to happen
+     * @param listener {@link ActionListener} which will be called when prediction is completed or errored out.
+     */
+    public void inferenceSentences(
+        @NonNull final List<String> targetResponseFilters,
+        @NonNull final String modelId,
+        @NonNull final List<String> inputText,
+        @NonNull final ActionListener<List<List<Float>>> listener
+    ) {
+        final ModelResultFilter modelResultFilter = new ModelResultFilter(false, true, targetResponseFilters, null);
+        final MLInputDataset inputDataset = new TextDocsInputDataSet(inputText, modelResultFilter);
+        final MLInput mlInput = new MLInput(FunctionName.CUSTOM, null, inputDataset, MLModelTaskType.TEXT_EMBEDDING);
```

Comment:
We plan to remove the CUSTOM function as we are going to support both custom and built-in text embedding models. The ml-commons latest code changed to  `TEXT_EMBEDDING`.

I'm ok to change in next PR

Reply:
Yeah sure I will change this in the next PR. Thanks for providing the update. This was as per the current interface that you provided.